src/ast/AstNode.ts

```diff
@@ -0,0 +1,8 @@
+export abstract class AstNode {
+
```

Comment:
nit: would prefer if this is an interface unless we have a specific reason for making this an abstract class (like we're planning to include some implementation here). 
Reasons: we can't easily extend multiple classes in typescript (and there may be a case we want to do this at some point?), and if we're just implementing this abstract class, then it's just treated as an interface anyways.

Reply:
@kiyomih thanks for your feedback!

I agree that usually, behaviour is better specified as an interface. I was thinking that we have some of the common classes, e.g. `Tokenizer`, `FileSystem`, live inside the abstract `AstNode` class, much like how Elisa implemented tinyDot in class. 

I'll wait for your response before I merge; architectural designs like these are important, thanks for bringing it up!

Reply:
That's a fair point! Everything looks good to me in that case.